js/pages/tola_management_pages/user/components/edit_user_programs.js

```diff
@@ -258,17 +259,39 @@ export default class EditUserPrograms extends React.Component {
                 return {...country, role: new_val, has_access: false}
             }
         })()
+        // using the form of setstate that receives the previous state/props as an argument, as we need
+        // to update the programs (based on current state programs list and new country value)
+        this.setState((state, props) => {
+            // this is a reference to the _old_ state programs, so it's safe to modify:
+            let statePrograms = state.programs;
```

Comment:
Any reason not to just use `state.programs` throughout, rather than setting up the reference?

Reply:
that's why we have code review 🤦  - thanks!